File: mycuration/src/main/java/com/phicdy/mycuration/alarm/BootReceiver.java
package com.phicdy.mycuration.alarm;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import com.phicdy.mycuration.util.PreferenceHelper;

public class BootReceiver extends BroadcastReceiver {
    public BootReceiver() {
    }

    @Override
    public void onReceive(Context context, Intent intent) {
        if (intent != null && intent.getAction().equals(Intent.ACTION_BOOT_COMPLETED)) {
            AlarmManagerTaskManager manager = new AlarmManagerTaskManager(context);
            PreferenceHelper helper = PreferenceHelper.INSTANCE;
            int intervalSec = helper.getAutoUpdateIntervalSecond();
            manager.setNewAlarm(intervalSec);
        }
    }
}

File: mycuration/src/main/java/com/phicdy/mycuration/domain/alarm/BootReceiver.kt
package com.phicdy.mycuration.domain.alarm

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent

import com.phicdy.mycuration.util.PreferenceHelper

class BootReceiver : BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent?) {
        if (intent != null && intent.action == Intent.ACTION_BOOT_COMPLETED) {
            AlarmManagerTaskManager(context).setNewAlarm(PreferenceHelper.autoUpdateIntervalSecond)
        }
    }
}
